containers/src/impl/Kokkos_UnorderedMap_impl.hpp

```diff
@@ -27,6 +27,22 @@
 namespace Kokkos {
 namespace Impl {
 
+//! Either append to the label if the property already exists, or set it.
+template <typename... P>
+auto with_updated_label(const ViewCtorProp<P...>& view_ctor_prop,
+                        const std::string& label) {
+  using vcp_t = ViewCtorProp<P...>;
+  //! If the label property is already set, append. Otherwise, set label.
+  if constexpr (vcp_t::has_label) {
+    vcp_t new_ctor_props(view_ctor_prop);
+    static_cast<ViewCtorProp<void, std::string>&>(new_ctor_props)
+        .value.append(label);
+    return new_ctor_props;
+  } else {
+    return Impl::with_properties_if_unset(view_ctor_prop, label);
+  }
+}
```

Comment:
AFAICT the label exists whenever this function is called.
```suggestion
//! Append to the label.
template <typename... P>
auto append_to_label(const ViewCtorProp<P...>& view_ctor_prop,
                        const std::string& label) {
  static_assert (vcp_t::has_label); {
  ViewCtorProp<P...> new_ctor_props(view_ctor_prop);
  static_cast<ViewCtorProp<void, std::string>&>(new_ctor_props)
        .value.append(label);
  return new_ctor_props;
}
```

Reply:
Feel free to push to my branch (make sure you format it properly)